pkg/storagev2/storage.go

```diff
@@ -71,8 +71,11 @@ type Query struct {
 	// to construct new object for the result.
 	Factory func() Item
 
-	// ItemAttribute indicates a specific interest of an Item attribute.
-	ItemAttribute ItemAttribute
+	// Prefix indicates interest in Item which hast this prefix.
+	Prefix string
```

Comment:
It should be specified here that this Prefix is for the ID and not Namespace.

Also, like in shed, we could potentially need a boolean `includeFirst` or something. This is to tell the iteration if we should start from the Prefix match item or the next one. Mainly, I am thinking of the pagination issue. So if you process some entries and then call Iterate a second time, usually this prefix would be the key of the last item you looked at. So this boolean would tell us that we should start from the next one. As opposed to esad's case where we need to include the first one as well in some cases.

Reply:
Done.